Taarafo.Core.Tests.Unit/Services/Foundations/Comments/CommentServiceTests.Exceptions.RetrieveById.cs

```diff
@@ -0,0 +1,98 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using EFxceptions.Models.Exceptions;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using System;
+using System.Threading.Tasks;
+using Taarafo.Core.Models.Comments;
+using Taarafo.Core.Models.Comments.Exceptions;
+using Xunit;
+
+namespace Taarafo.Core.Tests.Unit.Services.Foundations.Comments
+{
+    public partial class CommentServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowCriticalDependencyExceptionOnRetrieveByIdIfSqlErrorOccursAndLogItAsync()
+        {
+            // given
+            Guid someId = Guid.NewGuid();
+            SqlException sqlException = GetSqlException();
+
+            var failedCommentStorageException =
+                new FailedCommentStorageException(sqlException);
+
+            var expectedCommentDependencyException =
+                new CommentDependencyException(failedCommentStorageException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectCommentByIdAsync(It.IsAny<Guid>()))
+                    .ThrowsAsync(sqlException);
+
+            // when
+            ValueTask<Comment> retrieveCommentByIdTask =
+                this.commentService.RetrieveCommentByIdAsync(someId);
+
+            // then
+            await Assert.ThrowsAsync<CommentDependencyException>(() =>
+                retrieveCommentByIdTask.AsTask());
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectCommentByIdAsync(It.IsAny<Guid>()),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogCritical(It.Is(SameExceptionAs(
+                    expectedCommentDependencyException))),
+                        Times.Once);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
```

Comment:
Please change the hierarchy here - Storage, Logging, DateTime. Also, verify in other tests too.

Reply:
@ShriHumrudha, so just to confirm, you only want the order to be storage, datetime, logger on the happy path and then storage, logger, datetime on the exception tests as I changed the order around based on the previous request?